fe/fe-core/src/main/java/com/starrocks/analysis/SlotDescriptor.java

```diff
@@ -282,6 +283,11 @@ public TSlotDescriptor toThrift() {
         tSlotDescriptor.setIsMaterialized(true);
         tSlotDescriptor.setIsOutputColumn(isOutputColumn);
         tSlotDescriptor.setIsNullable(isNullable);
+
+        if (parent.getTable() != null && column != null && parent.getTable().isDeltalakeTable()) {
+            DeltaLakeTable table = (DeltaLakeTable) parent.getTable();
+            table.doColumnMapping(tSlotDescriptor, column);
```

Comment:
I think we should set this in `Column` and it will init `SlotDescriptor` at LOC:270

Reply:
Done